File: Equipments.Domain/Entities/EquipmentPositionHistory.cs

namespace Equipments.Domain.Entities
{
    public class EquipmentPositionHistory
    {
        public Guid EquipmentId { get; private set; }
        public DateTime Date { get; private set; }
        public long Latitude { get; private set; }
        public long Longitude { get; private set; }

        public EquipmentPositionHistory(Guid equipmentId, DateTime date, long latitude, long longitude)
        {
            EquipmentId = equipmentId;
            Date = date;
            Latitude = latitude;
            Longitude = longitude;
        }

        public void EditPosition(long latitude, long longitude) 
        {
            Latitude = latitude;
            Longitude = longitude;
            Date = DateTime.Now;
        }
    }
}

File: equipments/src/main/java/com/br/raf/equipments/entities/EquipmentPositionHistory.java
package com.br.raf.equipments.entities;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Objects;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@IdClass(EquipmentHistoryId.class)
@Table(name = "equipment_position_history")
public class EquipmentPositionHistory {
    @Id
    @Column(name = "equipment_id")
    private UUID equipmentId;
    @Id
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime date;
    private double lat;
    private double lon;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EquipmentPositionHistory that = (EquipmentPositionHistory) o;
        return equipmentId.equals(that.equipmentId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(equipmentId);
    }
}
